src/module.c

```diff
@@ -4314,7 +4326,8 @@ int RM_ListInsert(RedisModuleKey *key, long index, RedisModuleString *value) {
 int RM_ListDelete(RedisModuleKey *key, long index) {
     if (moduleListIteratorSeek(key, index, REDISMODULE_WRITE)) {
         listTypeDelete(key->iter, &key->u.list.entry);
-        moduleDelKeyIfEmpty(key);
+        if (moduleDelKeyIfEmpty(key)) return REDISMODULE_OK;
+        listTypeTryConvertQuicklist(key->value, moduleFreeListIterator, key);
```

Comment:
Is it possible to let listTypeDelete handle the conversion from quicklist to listpack?

I can be a better abstraction.